### PR TITLE
Tweak the introduction to akka persistence

### PR DESCRIPTION
**Pull request by dwijnand**
Specifically:

* avoid "internal", as I initially though that meant the Akka framework's internal state of an actor
* dropped "started" in the context of recovering an actor: in order to recover you have to have been already started
* rather than referring to "changes", reference that what's persisted are the actor's received events
* some general English re-wording

### CODE REVIEW

**Comment by dwijnand** [comment]
Review by @helena?

File: akka-docs/src/main/paradox/persistence.md
Comment (johanandren) [inline akka-docs/src/main/paradox/persistence.md]: You could also have an app that explicitly stops and starts persistent actors when they are needed. I think `started` should stay.

File: akka-docs/src/main/paradox/persistence.md
Comment (johanandren) [inline akka-docs/src/main/paradox/persistence.md]: It should not be the messages it receives (that'd be command sourcing) it is events representing state changes.

**Review by johanandren — COMMENTED** [review]
Good improvement, had some, you know, opinions.

File: akka-docs/src/main/paradox/persistence.md
Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: Oh right. lol

File: akka-docs/src/main/paradox/persistence.md
Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: I see.  I'll add it back.

Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: I didn't.  But I included your stop-start scenario, and loosened the wording to indicate these are just examples of things that can lead to a restart and recovery.

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-docs/src/main/paradox/persistence.md
Comment (johanandren) [inline akka-docs/src/main/paradox/persistence.md]: The actor receives commands (`ChangeTheName(newName)`) which are then validated and events (`NameChanged(newName)`) are created and those are persisted, so maybe "_events_ representing changes that happened to the actor state" or something like that?

Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: Do all events represent changes in state?

What is it you're trying to achieve with your wording?

Comment (ignasi35) [reply akka-docs/src/main/paradox/persistence.md]: @dwijnand I think the wording is a bit dense. How about:


```
Akka persistence enables stateful actors to persist their state so that it can be recovered when an actor is either restarted, such as after a JVM crash, by a supervisor or a manual stop-start, or migrated within a cluster. The key concept behind Akka persistence is that it's not the actor's state that is persisted but each change on the state. These state changes are called events. The events are persisted by appending to storage (no data in the storage is ever mutated) which allows for very high transaction rates and efficient replication. A stateful actor is recovered by replaying the stored events to the actor, allowing it to rebuild its state. This reconstruction operation is called rehydration. The rehydration can be either the full replay of events or starting from a snapshot (when available) dramatically reducing recovery times. 
```
I tried to focus on the concepts (e.g. removed the use of `receive`) and introduced some common language gradually (events, rehydration,...).

Finally, I don't know what the last sentence means:
```
Akka persistence also provides point-to-point communication with at-least-once message delivery semantics.
```

Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: Now you got me thinking it shouldn't say "enables stateful actors to persist their state" either...

Comment (ignasi35) [reply akka-docs/src/main/paradox/persistence.md]: >  "enables stateful actors to persist their state" 

Hmmm, good point. I guess we could say:

```
Akka persistence adds durability to stateful actors so that the state can 
be recovered when an actor is either restarted, such as after a JVM crash, by a supervisor or a manual stop-start, or migrated within a cluster.
```
WDYT?

Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: I think it's better.  But perhaps it's worth add a short definition of "durability".

Also, I've been thinking that it might be better (in your previous update) to lead with technical terms first and backfill the meaning.  So instead of

> A stateful actor is recovered by replaying the stored events to the actor, allowing it to rebuild its state. This reconstruction operation is called rehydration.

something like

> A stateful actor is recovered (by/with/via) rehydration, (that is?) by replaying the stored events to the actor allowing it to rebuild its state.

I'm not sure if it's a good idea.

Comment (ignasi35) [reply akka-docs/src/main/paradox/persistence.md]: SGTM

Comment (johanandren) [reply akka-docs/src/main/paradox/persistence.md]: > What is it you're trying to achieve with your wording?

Get the introduction to use event sourcing terminology but still explain in a clear way what it does (and not something it doesn't do).

I think I like @ignasi35 suggestion best so far.

> Finally, I don't know what the last sentence means:
>> Akka persistence also provides point-to-point communication 
>> with at-least-once message delivery semantics.

Akka Persistence contains the `AtLeastOnceDelivery` abstraction that does what it says on top of persistence.

Comment (johanandren) [reply akka-docs/src/main/paradox/persistence.md]: One thought though is that even if I have seen the term `rehydrating` in event sourcing contexts, the persistence APIs and existing docs use the terms `recover` the state by `replay` of the events, so maybe we should stick with those.

Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: I didn't specify which Monday.  ⌛️

File: akka-docs/src/main/paradox/persistence.md
Comment (helena) [inline akka-docs/src/main/paradox/persistence.md]: I might prefer to keep the 'internal' just to retain actor private / internal state vs being exposed by design. WDYT?

File: akka-docs/src/main/paradox/persistence.md
Comment (helena) [inline akka-docs/src/main/paradox/persistence.md]:
```suggestion
events and allowing the actor to rebuild its state. This can be either the full history of changes
```

File: akka-docs/src/main/paradox/persistence.md
Comment (helena) [inline akka-docs/src/main/paradox/persistence.md]:
```suggestion
(though actor state snapshot support is also available). Events are persisted to storage by an append-only log which
```

File: akka-docs/src/main/paradox/persistence.md
Comment (helena) [inline akka-docs/src/main/paradox/persistence.md]: ?
```suggestion
or starting from a checkpoint in a snapshot which can dramatically reduce recovery times. Akka persistence also provides point-to-point
```

**Review by helena — CHANGES_REQUESTED** [review]

File: akka-docs/src/main/paradox/persistence.md
Comment (dwijnand) [reply akka-docs/src/main/paradox/persistence.md]: My motivation for avoiding "internal" was that I initially thought "internal" meant the Akka framework's internal state of an actor or something, rather than "user-land" internal state.  Maybe it's just me?

**Review by helena — APPROVED** [review]

**Comment by dwijnand** [comment]
Thanks for the ✅!  I'll hopefully get to this today, but I won't mind either it landing as is or someone last mile-ing it.

File: akka-docs/src/main/paradox/persistence.md
Comment (patriknw) [reply akka-docs/src/main/paradox/persistence.md]: I think it's better to remove internal for the reasons mentioned.

**Review by patriknw — APPROVED** [review]
I think this is looking good. Is it ready to merge, or is it pending another iteration of improving the wording?

**Comment by dwijnand** [comment]
I'm happy to get this off my conscience and invite @johanandren and @ignasi35 to propose the good suggestions they made in https://github.com/akka/akka/pull/25916#pullrequestreview-174982831.

TL;DR ready to merge for me

**Review by ignasi35 — APPROVED** [review]

**Comment by patriknw** [comment]
@dwijnand I don't dare to click the Squash and merge button because this has "helena and dwijnand committed", could you squash manually and then we'll merge

**Comment by dwijnand** [comment]
Sure!  What happens if you squash and merge?

**Comment by patriknw** [comment]
We have seen that Github assigns attribution to wrong contributor. Not sure if it was exactly this case. I think it used the person of the last commit, while we would prefer person on first commit. They changed the behavior back and forth, I thinks. So I lost trust in that button and never use it when I see more than one person in the commit list.

Similar problems with the UI for resolving conflicts.

**Comment by dwijnand** [comment]
OK, good to know.  I've not used GitHub's squash.

**Pull request merged**
